cognite/client/utils/_identifier.py

```diff
@@ -226,3 +239,31 @@ class SingletonIdentifierSequence(IdentifierSequenceCore[Identifier]):
 
 class DataModelingIdentifierSequence(IdentifierSequenceCore[DataModelingIdentifier]):
     ...
+
+
+class UserIdentifierSequence(IdentifierSequenceCore[UserIdentifier]):
+    @classmethod
+    def load(cls, user_identifiers: str | Sequence[str]) -> UserIdentifierSequence:
+        if isinstance(user_identifiers, str):
+            return cls(identifiers=[UserIdentifier(user_identifiers)], is_singleton=True)
+
+        elif isinstance(user_identifiers, Sequence):
+            return cls(identifiers=list(map(UserIdentifier, map(str, user_identifiers))), is_singleton=False)
+
+        raise TypeError(f"user_identifiers must be of type str or Sequence[str]. Found {type(user_identifiers)}")
+
+    def chunked(self, chunk_size: int) -> list[UserIdentifierSequence]:  # type: ignore [override]
+        return [
+            UserIdentifierSequence(chunk, is_singleton=self.is_singleton())
+            for chunk in split_into_chunks(self._identifiers, chunk_size)
+        ]
+
+    def assert_singleton(self) -> None:
+        if not self.is_singleton():
+            raise ValueError("Exactly one user identifier (string) must be specified")
```

Comment:
If the parent class take in `identifier name` which cold be used in this error message, then that would improve all the other child classes as well, and avoid this reimplementation.

Reply:
How would you write the error message to be generic?
```py
raise ValueError("Exactly one of id or external id must be specified")
raise ValueError("Exactly one user identifier (string) must be specified")
```

Reply:
I guess it would be the entire message, but it is more explicit to just overwrite.